connector/src/main/java/org/geysermc/connector/GeyserConnector.java

```
@@ -423,6 +443,44 @@ public boolean useXmlReflections() {
         return !this.getPlatformType().equals(PlatformType.FABRIC) && !"DEV".equals(GeyserConnector.VERSION);
     }
 
+    /**
+     * Register a Plugin Channel
+     *
+     * This will maintain what channels are registered and ensure new connections and existing connections
+     * are registered correctly
+     *
+     * @param channel Channel to register
+     */
+    public void registerPluginChannel(String channel) {
+        if (registeredPluginChannels.contains(channel)) {
+            return;
```

This should throw an exception instead of silently returning.